geomesa-convert/geomesa-convert-common/src/main/scala/org/locationtech/geomesa/convert/Transformers.scala

```diff
@@ -284,13 +286,33 @@ class IdFunctionFactory extends TransformerFunctionFactory {
   val string2Bytes = TransformerFn("string2bytes") { args => args(0).asInstanceOf[String].getBytes(StandardCharsets.UTF_8) }
 
   case object MD5 extends TransformerFn {
-
     override def name: String = "md5"
     val hasher = Hashing.md5()
-    override def eval(args: Any*): Any = hasher.hashBytes(args(0).asInstanceOf[Array[Byte]]).toString
+    override def eval(args: Any*)(implicit ctx: EvaluationContext): Any = hasher.hashBytes(args(0).asInstanceOf[Array[Byte]]).toString
   }
 
   val uuidFn = TransformerFn("uuid")   { args => UUID.randomUUID().toString }
   val base64 = TransformerFn("base64") { args => Base64.encodeBase64URLSafeString(args(0).asInstanceOf[Array[Byte]]) }
 
 }
+
+class FileInfoFunctionFactory extends TransformerFunctionFactory {
+  override def functions = Seq(FileNameFn(), LineNumberFn())
+
+  case class LineNumberFn() extends TransformerFn {
+    override def getInstance: LineNumberFn = LineNumberFn()
+    override def name: String = "lineNo"
+    override def eval(args: Any*)(implicit ec: EvaluationContext): Any = ec.incCount()
```

Comment:
This should just access the current line number, not increment it.  Increment the line number in the converter as it processes each line.